src/platforms/linux/daemon/apptracker.h

```diff
@@ -36,23 +24,26 @@ class AppTracker final : public QObject {
   void userCreated(uint userid, const QString& xdgRuntimePath);
   void userRemoved(uint userid);
 
-  QHash<QString, AppData*>::iterator begin() { return m_runningApps.begin(); }
-  QHash<QString, AppData*>::iterator end() { return m_runningApps.end(); }
-  AppData* find(const QString& cgroup) { return m_runningApps.value(cgroup); }
+  QStringList getRunningApps() const { return m_runningApps.keys(); }
+  QString getDesktopId(const QString& cgroup) const {
+    return m_runningApps.value(cgroup);
+  }
+  QStringList findByDesktopId(const QString& desktopId) const {
+    return m_runningApps.keys(desktopId);
```

Comment:
Is the cgroup path <--> desktopId mapping 1:1? Or can there be multiple cgroup paths that map to a desktopId?

Reply:
The desktopId -> cgroup mapping is one-to-many. Each cgroup matches at most one desktopId, but a desktopId can match multiple cgroups.